AlicerceTests/ServiceLocatorTests.swift

```diff
@@ -189,11 +383,129 @@ class ServiceLocatorTests : XCTestCase {
             XCTAssertEqual(rStringService, testLazyServiceString())
             
         } catch let error {
-            XCTAssert(false, "💥: Register failed with error: \(error)")
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    // MARK: Failure
+
+    func testGet_UsingTypeInference_ShouldFailWithInexistingService() {
+
+        do {
+
+            // Try to fetch any service. Shouldn't return nothing.
+            let _: String = try serviceLocator.get()
+
+        } catch  ServiceLocatorError.inexistentService {
+            // Expected result
+        } catch {
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    func testGet_UsingName_ShouldFailWithInexistingService() {
+
+        let serviceName = "TestServiceName"
+
+        do {
+
+            // Try to fetch any service. Shouldn't return nothing.
+            let _: String = try serviceLocator.get(name: serviceName)
+
+        } catch  ServiceLocatorError.inexistentService {
+            // Expected result
+        } catch {
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    func testGet_UsingTypeInference_ShouldFailWithServiceTypeMismatch() {
+
+        let testService = ""
+        let serviceName = "Double"
+
+        do {
+
+            // Register a String type service
+            try serviceLocator.register(name: serviceName, service: testService)
+
+            // Get a Double type service
+            let _: Double = try serviceLocator.get()
+
+        } catch ServiceLocatorError.serviceTypeMismatch(expected: let expectedType, found: let foundType) {
+            XCTAssertTrue(type(of: expectedType) != type(of: foundType))
+        } catch {
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    func testGet_UsingName_ShouldFailWithServiceTypeMismatch() {
+
+        let testService = ""
+        let serviceName = "TestServiceName"
+
+        do {
+
+            // Register a String type service
+            try serviceLocator.register(name: serviceName, service: testService)
+
+            // Get a Double type service
+            let _: Double = try serviceLocator.get(name: serviceName)
+
+        } catch ServiceLocatorError.serviceTypeMismatch(expected: let expectedType, found: let foundType) {
+            XCTAssertTrue(type(of: foundType) != type(of: expectedType))
```

Comment:
The equality test on `expectedType` (`Double` in this case) is to test if the error type itself is well built, so I think it adds value.

By only testing against `testService`'s type, we aren't testing this.

Besides, the test `expectedType != type(of: testService)` can already be logically implied from the other two tests:

`expectedType != foundType` **and**  `expectedType == type(of: testService)` **implies** `foundType != type(of: testService)` 🤓 

Don't you agree?